tests/test_class.py

```diff
@@ -329,3 +329,21 @@ class PyNonFinalFinalChild(m.IsNonFinalFinal):
 def test_exception_rvalue_abort():
     with pytest.raises(RuntimeError):
         m.PyPrintDestructor().throw_something()
+
+
+def test_pyexception_destructor():
+    import sys
+
+    if hasattr(sys, "unraisablehook"):
```

Comment:
Only Python 3.8+ have `unraisablehook`. I think for earlier versions we could call a subprocess and check its stderr.

Reply:
That's why I made it conditional. If the logic works on one version of python since this is so esoteric I didn't think it was worth testing on the older versions too.

Reply:
pybind11 has some way of capturing the stderr. It would definitely be good to test on all Python versions, if possible (especially Python 2, which tends to act up/do things differently!).